### PR TITLE
Create depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu

### PR DESCRIPTION
**Pull request by indugadiraju**
Hi Dr. B! I have uploaded and committed my changes to this file! Please let me know if you are able to see them now.

### CODE REVIEW

File: Firmware/depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+    // approximately every second or so, print out the current stats
+    if (millis() - timer > 1000) {
+        timer = millis(); // reset the timer
+
+        serialPrintGPSTime();
+
+        temperatureC = getTemp();
+        JSN_SR04_Gen3::DistanceResult result = distanceSesnor.getDistance();
+        depth = distanceCallback(result);
```
Comment (indugadiraju) [inline Firmware/depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu:116]: Here, in the main loop, I have called distanceCallback() function to get the depth in cm. Later, in line 129, when printToFile() is called, the depth will be saved and written onto the SD card.

File: Firmware/depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    dataFile.print(GPS.lon); // E or W
+    dataFile.print(",");
+
+    // Temperature
+    dataFile.print(temperatureC);
+    dataFile.print(",");
+
+    // Depth/distance
+    dataFile.print(depth);
+    dataFile.print(",");
```
Comment (indugadiraju) [inline Firmware/depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu:252]: Here, I am writing "depth," which had received a value in the main loop, to the SD card.

**Comment by SUPScientist** [comment]
@InduGadi, nice work, I've merged this into the main branch. There were several issues still which I tried to resolve (check out the most recent commits here: https://github.com/SUPScientist/smart-coasts-bathy-mapping/commits/main), most notably that your new code wasn't in a Particle project, so I created a new project, put your code in it, added the libraries, made a few tweaks, and compiled. It still doesn't work perfectly (the depth just shows "0.00" frequently) but we'll have to troubleshoot that separately.

**Comment by indugadiraju** [comment]
Thanks Dr. B! I'm sorry about all of the issues in the code. I checked the recent commits and they look great! Do you think the depth readings showing 0.00 is a firmware or hardware issue? Also, is everything saving to the SD card consistently?
Thank you so much, and have a happy Thanksgiving!

**Comment by SUPScientist** [comment]
Happy Thanksgiving to you, too. No need to apologize! It's a learning process for all of us, me included. I just wanted to point out the changes I made for the record and for the educational value. 

Re: 0.00 hardware vs. firmware... I'm not sure, I had the same question. I'm opening an issue (over here: https://github.com/SUPScientist/smart-coasts-bathy-mapping/issues/7) to document the problem and approach that I will try to take. If you come across anything on the firmware side of things, please let me know!

**Comment by indugadiraju** [comment]
That sounds great Dr. B! On my end, I will look at any potential firmware issues that could be causing the 0.00 problem.

**Pull request merged**